src/shared/wmistr.rs

```diff
@@ -0,0 +1,203 @@
+pub const WNODE_FLAG_PDO_INSTANCE_NAMES: ULONG = 0x00010000;
+pub const WNODE_FLAG_TRACED_GUID: ULONG = 0x00020000;
+pub const WNODE_FLAG_LOG_WNODE: ULONG = 0x00040000;
+pub const WNODE_FLAG_USE_GUID_PTR: ULONG = 0x00080000;
+pub const WNODE_FLAG_USE_MOF_PTR: ULONG = 0x00100000;
+pub const WNODE_FLAG_NO_HEADER: ULONG = 0x00200000;
+pub const WNODE_FLAG_SEND_DATA_BLOCK: ULONG = 0x00400000;
+pub const WNODE_FLAG_VERSIONED_PROPERTIES: ULONG = 0x00800000;
+pub const WNODE_FLAG_SEVERITY_MASK: ULONG = 0xff000000;
+STRUCT!{struct OFFSETINSTANCEDATAANDLENGTH {
+    OffsetInstanceData: ULONG,
+    LengthInstanceData: ULONG,
+}}
+pub type POFFSETINSTANCEDATAANDLENGTH = *mut OFFSETINSTANCEDATAANDLENGTH;
+UNION2!{union WNODE_ALL_DATA_u {
+    [u8; 8],
+    FixedInstanceSize FixedInstanceSize_mut: ULONG,
+    OffsetInstanceDataAndLength OffsetInstanceDataAndLength_mut:
+        [OFFSETINSTANCEDATAANDLENGTH; ANYSIZE_ARRAY],
+}}
+STRUCT!{struct WNODE_ALL_DATA {
+    WnodeHeader: WNODE_HEADER,
+    DataBlockOffset: ULONG,
+    OffsetInstanceNameOffsets: ULONG,
+    u: WNODE_ALL_DATA_u,
+}}
+pub type PWNODE_ALL_DATA = *mut WNODE_ALL_DATA;
+STRUCT!{struct WNODE_SINGLE_INSTANCE {
+    WnodeHeader: WNODE_HEADER,
+    OffsetInstanceName: ULONG,
+    InstanceIndex: ULONG,
+    DataBlockOffset: ULONG,
+    SizeDataBlock: ULONG,
+    VariableData: [UCHAR; ANYSIZE_ARRAY],
+}}
+pub type PWNODE_SINGLE_INSTANCE = *mut WNODE_SINGLE_INSTANCE;
+STRUCT!{struct WNODE_SINGLE_ITEM {
+    WnodeHeader: WNODE_HEADER,
+    OffsetInstanceName: ULONG,
+    InstanceIndex: ULONG,
+    ItemId: ULONG,
+    DataBlockOffset: ULONG,
+    SizeDataItem: ULONG,
+    VariableData: [UCHAR; ANYSIZE_ARRAY],
+}}
+pub type PWNODE_SINGLE_ITEM = *mut WNODE_SINGLE_ITEM;
+STRUCT!{struct WNODE_METHOD_ITEM {
+    WnodeHeader: WNODE_HEADER,
+    OffsetInstanceName: ULONG,
+    InstanceIndex: ULONG,
+    MethodId: ULONG,
+    DataBlockOffset: ULONG,
+    SizeDataBlock: ULONG,
+    VariableData: [UCHAR; ANYSIZE_ARRAY],
+}}
+pub type PWNODE_METHOD_ITEM = *mut WNODE_METHOD_ITEM;
+STRUCT!{struct WNODE_EVENT_ITEM {
+    WnodeHeader: WNODE_HEADER,
+}}
+pub type PWNODE_EVENT_ITEM = *mut WNODE_EVENT_ITEM;
+UNION2!{union WNODE_EVENT_REFERENCE_u {
+    [u8; 4],
+    TargetInstanceIndex TargetInstanceIndex_mut: ULONG,
+    TargetInstanceName TargetInstanceName_mut: [WCHAR; ANYSIZE_ARRAY],
+}}
+STRUCT!{struct WNODE_EVENT_REFERENCE {
+    WnodeHeader: WNODE_HEADER,
+    TargetGuid: GUID,
+    TargetDataBlockSize: ULONG,
+    u: WNODE_EVENT_REFERENCE_u,
+}}
+pub type PWNODE_EVENT_REFERENCE = *mut WNODE_EVENT_REFERENCE;
+STRUCT!{struct WNODE_TOO_SMALL {
+    WnodeHeader: WNODE_HEADER,
+    SizeNeeded: ULONG,
+}}
+pub type PWNODE_TOO_SMALL = *mut WNODE_TOO_SMALL;
+#[cfg(target_arch = "x86")]
+const __SIZE_OF_WMIREGGUIDW_u: usize = 4;
+#[cfg(target_arch = "x86_64")]
+const __SIZE_OF_WMIREGGUIDW_u: usize = 8;
+UNION2!{union WMIREGGUIDW_u {
+    [u8; __SIZE_OF_WMIREGGUIDW_u],
```

Comment:
I recently added support for `UNION2` to have two storages specified, one for 32bit and one for 64bit. Also, an array of `u8` is usually the wrong thing because the union needs to have the correct alignment (otherwise when placed inside a struct it could have the wrong position resulting in issues like https://github.com/retep998/winapi-rs/issues/452).